### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by galanovnick**
This PR provides the initial implementation of the mixin which is intended to detect element theme.

@kompike  PTAL.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## SpineContraster
+
+`SpineContraster` determines to which dark or light theme an element belongs based on its
+background color. Sets `darkTheme` and `lightTheme` boolean properties to the element.
```
Comment (kompike) [inline README.md:4]: As for me, `determines the theme (dark or light) an element belongs to based on its background color`, sounds a little bit better.

File: bower.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "spine-contraster",
+  "description": "A mixin to determine to which dark or light theme an element belongs based on its background color.",
```
Comment (kompike) [inline bower.json]:
```suggestion
  "description": "A mixin to determine the theme (dark or light) an element belongs to based on its background color.",
```

File: demo/index.html [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                    </div>
+                </template>
+            </dom-bind>
+
+            <script>
+                const autobind = document.body.querySelector('#autobind');
+                autobind.colors = randomColors();
+                autobind.randomizeColors = () => autobind.colors = randomColors();
+
+                function randomColors() {
```
Comment (kompike) [inline demo/index.html]: Maybe it is better to name it `getRandomColors`?

File: demo/index.html [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                    const colors = [];
+                    for (let i = 0 ; i < 20; i++) {
+                        colors.push(`#${randomHex()}${randomHex()}${randomHex()}`)
+                    }
+                    return colors;
+                }
+
+                function randomHex() {
```
Comment (kompike) [inline demo/index.html]: Maybe it is better to name this function `getRandomHex`?

File: demo/index.html [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    <link rel="import" href="./spine-contraster-demo.html">
+
+    <custom-style>
+        <style is="custom-style" include="demo-pages-shared-styles">
+        </style>
+    </custom-style>
+</head>
+<body>
+<div class="vertical-section-container centered">
+    <h3>An element with SpineContrastorMixin can detect its theme</h3>
```
Comment (kompike) [inline demo/index.html]:
```suggestion
    <h3>An element with SpineContrastorMixin can detect its own theme</h3>
```

File: spine-contraster.html [hunk 6]
```diff
@@ -0,0 +1,152 @@
+<link rel="import" href="../polymer/polymer-element.html">
+
+<script>{
+    /**
+     * `SpineContraster` is a mixin to determine to which dark or light theme an element belongs
```
Comment (kompike) [inline spine-contraster.html]: Let's use the same as for `README.md`, `determines the theme (dark or light) an element belongs to based on its background color`, what do you think?

File: spine-contraster.html [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+                for (let i = 0; i < colors.length; i++) {
+                    const color = colors[i];
+                    tokens[color] = colorRgb[color] / 255;
+                    if (tokens[color] <= 0.03928) {
+                        tokens[color] = tokens[color] / 12.92;
+                    } else {
+                        tokens[color] = Math.pow((tokens[color] + 0.055) / 1.055, 2.4);
+                    }
+                }
+                return 0.2126 * tokens.r + 0.7152 * tokens.g + 0.0722 * tokens.b;
```
Comment (kompike) [inline spine-contraster.html:88]: Is it possible to move all figures from `_calculateLuminance` function to constants?

**Review by kompike — CHANGES_REQUESTED** [review]
@galanovnick PTAL

**Comment by galanovnick** [comment]
@kompike PTAL

**Review by kompike — APPROVED** [review]
@galanovnick LGTM

**Pull request merged**